---
title: "Cvir_DNA_Meth_CpGOE"
author: "HM Putnam"
date: "02/28/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## load germline DNA methylation via calculation of CpGo/e for each gene in the C. virginica genome


## R Markdown

PCA for all CpG OE 
GLM by overall for mean CpG OE
GLM by population for mean CpG OE
GLM by GOslim term and 




```{Merge CpG with Annotation}

#Merge GO Slim annotation and CpG




```


```{Generate GO Slim Terms}
BP.Ids <- as.character(BP.INT$category)
myCollection <- GOCollection(BP.Ids)
fl <- "http://www.geneontology.org/ontology/subsets/goslim_generic.obo"
slim <- getOBOCollection(fl, evidenceCode="TAS")
BP.slim.INT <- goSlim(myCollection, slim, "BP")
BP.slim.INT <-BP.slim.INT[,c(1,3)]
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
